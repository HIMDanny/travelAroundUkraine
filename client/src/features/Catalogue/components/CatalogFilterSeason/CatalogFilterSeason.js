import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Box, Checkbox, FormControlLabel } from '@mui/material';
import { FilterAccordion } from '..';
import { setSeasons, setAllSeasons } from '../../../../store/slices/filterSlice/filterSlice';

const seasons = ['Summer', 'Autumn', 'Winter', 'Spring'];

const CatalogFilterSeason = () => {
  const [checked, setChecked] = useState(Array(seasons.length).fill(false));
  const dispatch = useDispatch();
  const filterSeasons = useSelector((store) => store.filter.seasons);
 
  useEffect(() => {
    setChecked(() => {
      if (filterSeasons.length <= 0) {
        return Array(seasons.length).fill(false);
      }
      return seasons.map((season) => filterSeasons.indexOf(season) !== -1);
    });
  }, [filterSeasons]);


  const isAllChecked = checked.every((el) => el === true);

  const handleChangeAll = () => {
    setChecked((current) => {
      const newChecked = [...current];
      if (!isAllChecked) {
        newChecked.fill(true);
        dispatch(setAllSeasons(seasons));
      } else {
        newChecked.fill(false);
        dispatch(setAllSeasons([]));
      }
      return newChecked;
    });
  };

  const handleChange = async (event) => {
    const value = event.target.id;
    const index = seasons.indexOf(value);
    setChecked((current) => {
      const newChecked = [...current];
      newChecked[index] = !newChecked[index];
      return newChecked;
    });
    dispatch(setSeasons(value));
  };

  const children = (
    <Box sx={{ display: 'flex', flexDirection: 'column', ml: 3 }}>
      {seasons.map((el) => (
        <FormControlLabel
          key={el}
          label={el}
          control={<Checkbox id={el} checked={checked[seasons.indexOf(el)]} onChange={handleChange} />}
        />
      ))}
    </Box>
  );

  return (
    <FilterAccordion title="Season">
      <FormControlLabel label="All seasons" control={<Checkbox checked={isAllChecked} onChange={handleChangeAll} />} />
      {children}
    </FilterAccordion>
  );
};

export default CatalogFilterSeason;
