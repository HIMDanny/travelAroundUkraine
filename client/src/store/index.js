import { configureStore } from '@reduxjs/toolkit';
import userReducer from './slices/userSlice/userSlice';
import tourReducer from './slices/tourSlice/tourSlice';
import catalogueReducer from './slices/catalogueSlice/catalogueSlice';
import cartReducer from './slices/cartSlice/cartSlice';
import filterReducer from './slices/filterSlice/filterSlice';
import orderSlice from './slices/orderSlice/orderSlice';
import searchBarReducer from './slices/searchBarSlice/searchBarSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    tour: tourReducer,
    catalogue: catalogueReducer,
    cart: cartReducer,
    filter: filterReducer,
    order: orderSlice,
    searchBar: searchBarReducer,
  },
});

export default store;
