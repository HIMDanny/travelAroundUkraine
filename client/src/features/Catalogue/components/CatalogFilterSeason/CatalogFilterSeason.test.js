import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../store';
import CatalogFilterSeason from './CatalogFilterSeason';

jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Checkbox', () => ({ children }) => <span>{children}</span>);
jest.mock('@mui/material/FormControlLabel', () => ({ children }) => <label>{children}</label>);

jest.mock('../FilterAccordion/FilterAccordion', () => ({ children }) => <div>{children}</div>);

describe('CatalogFilterSeason snapshot test', () => {
  test('should CatalogFilterSeason match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CatalogFilterSeason />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
