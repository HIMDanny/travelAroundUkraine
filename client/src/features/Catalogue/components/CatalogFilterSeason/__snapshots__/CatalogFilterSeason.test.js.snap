// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogFilterSeason snapshot test should CatalogFilterSeason match snapshot 1`] = `
<DocumentFragment>
  <div>
    <label />
    <div>
      <label />
      <label />
      <label />
      <label />
    </div>
  </div>
</DocumentFragment>
`;
